compiler/rustc_passes/src/loops.rs

```diff
@@ -91,6 +91,10 @@ impl<'a, 'hir> Visitor<'hir> for CheckLoopVisitor<'a, 'hir> {
             }) => {
                 let cx = match kind {
                     hir::ClosureKind::Coroutine(hir::CoroutineKind::Desugared(kind, source)) => {
+                        let fn_decl_span = match self.cx {
```

Comment:
This seems like a bit too specific of a case here. I think we should just fix the `fn_decl_span` for the `hir::Closure` from an async closure.